solution_collector/src/main.rs

```diff
@@ -1,126 +1,202 @@
 #[macro_use]
 extern crate serde_json;
+#[macro_use]
+extern crate structopt;
 extern crate github_rs;
 extern crate regex;
+extern crate chrono;
 
 use std::io::Error;
-use std::fmt::Write;
+use std::fmt::Display;
+use std::fmt;
 
 use std::env;
 use github_rs::StatusCode;
 use github_rs::client::{Github, Executor};
 use serde_json::Value;
 use regex::Regex;
 use std::collections::HashMap;
+use structopt::StructOpt;
 
-#[derive(Debug)]
-enum Mode {
+use chrono::DateTime;
+use chrono::offset::Utc;
+
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "soco")]
+struct Opt {
+    #[structopt(subcommand)]
+    command: Command
+}
+
+#[derive(StructOpt, Debug)]
+enum Command {
+    #[structopt(name = "preview")]
     Preview,
-    Merge
+    #[structopt(name = "week-report")]
+    WeekReport {
+        week: u32
+    },
+    #[structopt(name = "merge")]
+    Merge {
+        week: u32,
+    }
+}
+
+struct Context {
+    owner: String,
+    repo: String,
+    client: Github
+}
+
+impl Context {
+    fn new(token: String, owner: String, repo: String) -> Self {
+        let client = Github::new(token).expect("Invalid token");
+        Context {
+            owner,
+            repo,
+            client
+        }
+    }
+}
+
+#[derive(Debug)]
+struct Entry {
+    title: String,
+    done: bool
+}
+
+#[derive(Debug)]
+struct WeekSubmitPR {
+    week: u32,
+    number: u32,
+    owner: String,
+    entries: Vec<Entry>,
+    created_at: DateTime<Utc>,
+}
+
+impl Display for WeekSubmitPR {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Owner: {:<20}. Entries: [", self.owner)?;
+        for entry in self.entries.iter() {
+            write!(f, "{}: ", entry.title)?;
+            if entry.done {
+                write!(f, "{:<2}", "X")?;
+            } else {
+                write!(f, "{:<2}", " ")?;
+            }
+        }
+        write!(f, "]. At: {}", self.created_at)
+    }
 }
 
-fn get_week(title: &str) -> u64 {
+fn get_week(title: &str) -> u32 {
```

Comment:
Should we name this funciton `parse_week`, and instead of returning `0` if we cannot parse, should we return `Option<u32>`?

Reply:
upvote for `parse_week`, so it would be consistent with other function names in this impl.